### ISSUE TITLE
Web - No Skip Navigation for All Pages

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#69]
There is no skip navigation link to be able to bypass the ILAB logo, DOL seal, Homepage link, and the drop-down menu.

**Comment by KyleLivingstonDOL** [comment]
John, are you referring to someone using Tab to skip over these? If we skip over these, is that a 508 issue? Because it is my understanding that we cannot skip over images in case a user who is hard of sight wants to get more information on what an image is, including our logo, seal, etc. Please advise.

**Comment by Abychalil** [comment]
Colton Daily added a comment - 16/Jul/19 12:11 AM (#269 in JIRA)

Need more info on the ask.

**Comment by jfc3-dol** [comment]
For those using a keyboard or assistive technology like a screen reader, a skip navigation link is needed, so a keyboard user doesn't have to tab through every link in the header section of the page and your menu. Also, the screen reader doesn't have to listen to all those items over and over again for each page in the application. By adding a skip link right after the BODY tag and then an anchor tag at the start of the main content section for each page helps improve the experience and accessibility for the user.

Here is a link to an example page of mine that uses a "Skip to Main Content" link which shows up when you tab to it. Once it shows either click the link or hit "Enter" on your keyboard and it will move you visually along with the keyboard focus because I also added to the in-page anchor a tabindex="-1" that fixes issues with a few browsers related to keyboard focus.

http://jfciii.com/presentations/examples/add.html

You probably will want to change the skip link background color, so it matches the looks of the website.

If you want next mid to later week maybe we can meet and go over any issues, you have questions on be it this one or others on either how to could it or show you with the keyboard or screen reader or even Android or iOS it that is an issue too.

**Comment by trevitalgit** [comment]
already implement in ticket
SDMISSSTC-268 508- Web - Home Page - Poor Color Contrast if Images Do Not Load #70

**Comment by jfc3-dol** [comment]
In Chrome, I do not see any indication that there is a skip link.

**Comment by Abychalil** [comment]
Fixing in progress

**Comment by jfc3-dol** [comment]
I do not see any skip navigation links for any web browsers I checked (IE11, Edge, Chrome, and Firefox).

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

I still do not see any skip navigation links for any web browsers I checked (IE11, Edge, Chrome, and Firefox).

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

Here is skip navigation CSS we talked about

CSS

```
.hide a, .hide a:hover, .hide a:visited {
display: block;
position: absolute;
left:-10000px;
top:auto;
width: 100%;
height: 3em;
font-weight: bold;
text-align: center;
}
.hide a:active, .hide a:focus {
background: #BEFF9E;
position: static;
width: auto;
height: auto;
padding: 1em;
margin: auto;
}
```

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

After Sarath reloaded the ILAB Comply Chain code and I hit the refresh combination a few times in the four browsers, the "Skip to Main Content" links show on the page.

I'm not sure if the CSS was not moved so it would be hidden until a person tabs to the link, then it would appear.

Please see the attached screenshot.


![ilab comply chain - home page - skip navigation link is showing on the page](https://user-images.githubusercontent.com/10423487/114922367-8fce1f80-9df9-11eb-8652-2336d4eb2f90.png)

 While testing the skip navigation function on secondary pages, the skip navigation link shows up when I tab to the "Comply Chain Homepage" link in multiple browsers.

Not sure why that is happening besides something is wrong with the CSS or missing.

I would expect it to show up as the first tab stop. The issue could be because of the HTML code order or missing/wrong CSS.

Also, the "Skip to Main Content" link is hidden on these secondary pages to start.

Please see the attached screenshot.

![ilab comply chain - step 1 page - skip navigation link does not show on the page until I tab to Comply Chain Homepage link](https://user-images.githubusercontent.com/10423487/114922453-aeccb180-9df9-11eb-8a6a-2e75d90eabac.png)

**Comment by jfc3-dol** [comment]
This issue is being worked on now to make a few changes, so it works as expected.

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

When using the skip navigation link in Firefox, it works the first time I use it on the home page, but when I use it on a secondary page, it visually moves me to the anchor while the keyboard focus does not go there. Sometimes I even have to tab backward to get the link to show up. I would expect my focus to start at the top of the page/the skip navigation link if it is the first piece of code for the page.

When using Edge or Chrome, after using the skip navigation link on secondary pages, it then starts tabbing from I am assuming where the skip link anchor is located and not at the top of the page as I would expect.

When using IE 11, after using the skip navigation link, I can not tab back to any of the items in the header section. Meaning, I can not get to the Comply Chain home link or open the "Menu" drop-down using a keyboard.

**Comment by jfc3-dol** [comment]
@pbhatt17 

When using Edge or Chrome, the Skip navigation link works as expected.

When I use IE 11, I'm not able to tab to the Skip navigation link at all, which means I'm tabbing forward or backward.

So I'm going to keep this issue open.

**Comment by jfc3-dol** [comment]
@pbhatt17 

Now able to use the Skip navigation link in IE 11, Edge, or Chrome, so I'm going to close the issue.

NOTE - As I think we mentioned in the last meeting, you were going to make the "Skip" navigation link visible with IE 11, and it is.
